spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveViews.scala

```diff
@@ -45,14 +47,17 @@ case class ResolveViews(spark: SparkSession) extends Rule[LogicalPlan] with Look
   protected lazy val catalogManager: CatalogManager = spark.sessionState.catalogManager
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-    case u@UnresolvedRelation(nameParts, _, _)
-      if catalogManager.v1SessionCatalog.isTempView(nameParts) =>
+    case u@UnresolvedRelation(nameParts, _, _) if isTempView(nameParts) =>
       u
 
     case u@UnresolvedRelation(parts@CatalogAndIdentifier(catalog, ident), _, _) =>
       loadView(catalog, ident)
         .map(createViewRelation(parts, _))
         .getOrElse(u)
+
+    case DropIcebergView(r@ResolvedIdentifier(catalog, ident), ifExists)
+      if isTempView(ident.asMultipartIdentifier) || !isViewCatalog(catalog) =>
+      DropView(r, ifExists)
```

Comment:
Why do we need to convert back to a non-Iceberg plan? Is it that we always convert in the pre-substitution batch? If so, is it reasonable to detect whether it is an Iceberg `ViewCatalog` before converting rather than converting back?